### ISSUE TITLE
Feature Request: CTCSS / DCS tone decoding.

### ISSUE THREAD

**Issue by ellisgeek** [AlexandreRouma/SDRPlusPlus#93]
At some point in the future™ it would be really helpful if the ability to decode CTCSS and DCS tones could be added. A) to locate the tone being used on a frequency and B) to squelch the audio output using the decoded tone.

This was available in SDR# via 2 plugins from rtl-sdr.ru.

P.S. SDR++ is incredibly impressive!

**Comment by AlexandreRouma** [comment]
Will keep in mind, not sure how hard it is to implement.

**Comment by rough316** [comment]
I would really like this CTCSS / PL tone decoding squelch as well. Trying to use the SDR on a crowded channel and this would be very helpful.

**Comment by hicaoc** [comment]
Also need to implement filtering, for CTCSS, if there is CTCSS in the signal, then the audio output needs to be this dumb audio filter out, otherwise the horn or headphones will have this low frequency, affecting the hearing.

**Comment by kieths** [comment]
I just installed and tested SDR++ today.  I was looking for this same feature (CTCSS).  I see that SDRSharp has a plugin for this ([link](http://rtl-sdr.ru/page/ctcss-detektor-shumopodavitel)), as well as other plugins.  I'm not see how to add plugins to SDR++ (?).

**Comment by AlexandreRouma** [comment]
SDR# plugins aren't compatible with SDR++

**Comment by craigerl** [comment]
Agreed, it's hard to use this with repeater systems which transmit a PL tone.

You can use sox to filter it out, but really inconvenient, and introduces some lag.

socat stdout udp-listen:7355 | play   -t raw -b 16 -e signed  -r 48k - -r 16k sinc -n 32767  180-8000

**Comment by AlexandreRouma** [comment]
Note that an optional audio high-pass has been added which can get rid of the PL tones.
Still can't squelch or decode it though.

**Comment by NazimkovMykola** [comment]
I understand this option is still not implemented?

**Comment by craigerl** [comment]
It is implemented - check "High pass filter", PL tone goes away, works great.

**Comment by agausmann** [comment]
> It is implemented - check "High pass filter", PL tone goes away, works great.

This is not the same as decoding the tone or using it to open squelch.

**Comment by drew4392-ops** [comment]
@AlexandreRouma Much appreciated!  CTCSS squelch is very helpful.  DCS would be also extremely helpful, if time permits. 

Thank you

**Comment by DeanCorso-DC** [comment]
Hello!

Thank you so much for adding subtone support to the program - it really makes a huge difference when working with signals. I’d like to toss in a few wishes for future functionality, things that would make the experience even smoother:

A button that, once a subtone is detected, automatically sets it as the squelch tone. That way you wouldn’t have to pick it manually every time - super helpful when you’re flipping between channels that use different subtones.

A frequency manager that can save channels complete with their subtones. This would take away all the headache of re-selecting the right tone every time you switch. Save a channel, and when you call it up, every parameter (including the subtone) just snaps into place.

When you add extra “**Radio**” tabs to monitor more channels, it would be great to have quick-access channel buttons somewhere in the top area where the frequency is displayed. Clicking them would instantly jump to that channel. Showing the frequency on or inside the button would be perfect. And as a cherry on top, it would be amazing if the button outline could indicate activity: gray when there’s no signal, green when there’s activity (DMR, FM, etc.).

Also, I’d really love it if the interface tweaks available via _Debug → Show demo window → Configuration → Style_ could be made permanent. Turning on **Dark** under **Colors** and setting **FrameRounding** to 12 (the max) gives the program a really nice look. It would be fantastic if the program could remember these settings between sessions, and even better if we could customize the colors ourselves within that dark theme.

Thanks again for all your work!
